src/freenet/node/diagnostics/NodeDiagnostics.java

```diff
@@ -0,0 +1,53 @@
+/* This code is part of Freenet. It is distributed under the GNU General
+ * Public License, version 2 (or at your option any later version). See
```

Comment:
Oh, one other thing, paging @ArneBab for this as well: should we put new code under GPLv3 or do we stay at GPLv2 for everything? I vaguely remember pieces of discussions about this but have completely forgotten any outcome. 😁

Reply:
"GPLv2 or later" sounds good. I wouldn’t change that without need (need to check every part of the code whether it is "or later"). If we make even one file GPLv3 or later, then the whole project must be GPLv3 or later, which would pull more checks and changes.

When we include code under Apache 2.0, we’ll have to move the project to GPLv3 or later, then we could think about moving individual files to GPLv3 or later.

Reply:
Alright, thanks!